subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/execution/ExecuteActionsTaskExecutorTest.groovy

```diff
@@ -15,12 +15,9 @@
  */
 package org.gradle.api.internal.tasks.execution
```

Comment:
I was *this* close to ripping the whole package of unit tests out. They are all just asserting the implementation line by line. The only thing that stopped me was not to make this PR bigger and the possibility that they cover things that are not covered by integration tests.

Reply:
Please make that separate PR.

Reply:
Or maybe rewrite the tests with much less mocking. They are currently overasserting so much. This is a anti-pattern I see throughout our code base. Sometimes it is because of badly designed production code, but often I feel like it is because of a misunderstanding of how tests should verify one thing at a time.